src/transform.js

```diff
@@ -0,0 +1,165 @@
+'use strict';
+const b = require('@babel/core');
+const fs = require('fs');
+
+// addSetupNode adds the setup() call to the beginning of the script.
+function addSetupNode() {
+  const setupNode = b.template.statement(`
+    setup({single_test: true, explicit_done: false});
+  `)();
+  return {
+    visitor: {
+      Program(path) {
+        path.node.body.unshift(setupNode);
+      },
+    },
+  };
+}
+
+const transformBoolMap = {
+  'shouldBeTrue': 'assert_true',
+  'shouldBeFalse': 'assert_false',
+
+};
+
+function transformShouldBeBool() {
+  return {
+    visitor: {
+      CallExpression(path) {
+        if (transformBoolMap.hasOwnProperty(path.node.callee.name)) {
+          path.node.callee.name = transformBoolMap[path.node.callee.name];
+          const newArgument =
+            b.template.expression(path.node.arguments[0].value)();
+          path.node.arguments[0] = newArgument;
+        }
+      },
+    },
+  };
+}
+
+const transformValueMap = {
+  'shouldBeNaN': {
+    name: 'assert_equals',
+    value: b.types.identifier('NaN'),
+  },
+
+  'shouldBeNull': {
+    name: 'assert_equals',
+    value: b.types.nullLiteral(),
+  },
+  'shouldBeNonNull': {
+    name: 'assert_not_equals',
+    value: b.types.nullLiteral(),
+  },
+
+  'shouldBeUndefined': {
+    name: 'assert_equals',
+    value: b.types.identifier('undefined'),
+  },
+  'shouldBeDefined': {
+    name: 'assert_not_equals',
+    value: b.types.identifier('undefined'),
+  },
+
+  'shouldBeZero': {
+    name: 'assert_equals',
+    value: b.types.numericLiteral(0),
+  },
+  'shouldBeNonZero': {
+    name: 'assert_not_equals',
+    value: b.types.numericLiteral(0),
+  },
+
+  'shouldBeEmptyString': {
+    name: 'assert_equals',
+    value: b.types.stringLiteral(''),
+  },
+};
+
+function transformShouldBeValue() {
+  return {
+    visitor: {
+      CallExpression(path) {
+        if (transformValueMap.hasOwnProperty(path.node.callee.name)) {
+          const newArgument1 = transformValueMap[path.node.callee.name].value;
```

Comment:
instead of arguments 0 and 1, can we use actual and expected? Below also

Reply:
Not sure I understand, I think actual and expected would be misleading variable names. The 0 and 1 refer to the position of the argument in the list of new arguments. But now that you mention it, I can move the definition of newArgument0 to be before newArgument1, which would make things more clear. Open to other name suggestions though.

Reply:
So to clarify my original comment, in all the cases within this function, the "expected" value is a constant. For example - for shouldBeZero, 0 is the expected value, and the value being compared to 0 is the "actual" value (eg: maybe the length of some list).

In some testing frameworks, there is a convention that one of the args is the actual value and the other is the expected one. This makes it easier to produce meaningful output like "assert failed, we got X but we expected Y". However, I can't tell whether js-test.js follows such a pattern, so maybe this is irrelevant.

Reply:
Still confused. This is a source file, doing the transformation from the old argument to the new argument. The test file is testTransform.js!